### PR TITLE
Issue/7253 drop use generic

### LINKED ISSUE #7253: Drop `use_generic` from `Resource.deserialize` method
This is blocked on when the agent refactor integration is done.
Spawned from this discussion https://github.com/inmanta/inmanta-core/pull/7101/files#r1491241944

- double check that this isn't used anymore + drop the  `use_generic` parameter from `Resource.deserialize` [method](https://github.com/inmanta/inmanta-core/blob/a3281a3abb7c760e7573000b72e3f59c3d8beb42/src/inmanta/resources.py#L338
)
- iso8 only!

### PR DESCRIPTION
**Pull request by FloLey**
# Description

Drop `use_generic` from `Resource.deserialize` method

closes #7253 

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [x] Attached issue to pull request
- [x] Changelog entry
- [ ] Type annotations are present
- [ ] Code is clear and sufficiently documented
- [x] No (preventable) type errors (check using make mypy or make mypy-diff)
- [x] Sufficient test cases (reproduces the bug/tests the requested feature)
- [ ] Correct, in line with design
- [ ] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )
- [ ] If this PR fixes a race condition in the test suite, also push the fix to the relevant stable branche(s) (see [test-fixes](https://internal.inmanta.com/development/core/tasks/build-master.html#test-fixes) for more info)

### CODE REVIEW

**Comment by Hugo-Inmanta** [comment]
The ticket mentions this should wait until the agent refactor integration is done, I don't think it's done yet

File: changelogs/unreleased/7253-remove-use-generic.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+description: Drop `use_generic` from `Resource.deserialize` method
+issue-nr: 7243
```
Comment (Hugo-Inmanta) [inline changelogs/unreleased/7253-remove-use-generic.yml:2]:
```suggestion
issue-nr: 7253
```

File: changelogs/unreleased/7253-remove-use-generic.yml [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+change-type: patch
```
Comment (Hugo-Inmanta) [inline changelogs/unreleased/7253-remove-use-generic.yml:3]: I think merge tool will complain if we make this a patch but we don't merge it in branches that allow patches ie iso 6 and iso7. I think this has to be a major ?

**Review by Hugo-Inmanta — CHANGES_REQUESTED** [review]

File: changelogs/unreleased/7253-remove-use-generic.yml (hunk 2 above)
Comment (FloLey) [reply changelogs/unreleased/7253-remove-use-generic.yml:3]: I don't think so but I might be mistaken. I think it only complains the other way around. @sanderr?

Comment (sanderr) [reply changelogs/unreleased/7253-remove-use-generic.yml:3]: It does complain under certain circumstances (not these). If the branches are still aligned, a patch is not allowed to diverge them. But the branches are already diverged so then anything goes.

**Review by Hugo-Inmanta — APPROVED** [review]

**Pull request closed without merging**